torch/distributed/_composable/fsdp/_fsdp_param_group.py

```diff
@@ -52,6 +59,11 @@ def __init__(
         self.all_gather_stream: torch.cuda.Stream = default_stream
         self.reduce_scatter_stream: torch.cuda.Stream = default_stream
         self.all_gather_state = AllGatherStateHolder()
+        self.post_forward_order: List[FSDPParamGroup] = []
+        self._post_forward_indices: List[int] = []
+        # Used to avoid mistargeted backward prefetches in the case that some
+        # module is used in forward but not in backward
+        self.expected_backward_unshard_count: int = 0
```

Comment:
maybe change ``expected_`` to ``estimated_`` before addressing the case of ``requires_grad=True`` but not part of ``loss.backward`` ?

[optional] I also spend some time to realize it's meant for remaining unfinished unshards since we are ``+/- = 1``. so I was thinking of ``estimated_unfinished_backward_unshard``